src/unitxt/metrics.py

```diff
@@ -1417,6 +1417,117 @@ def compute(
+    """
+
+    model_name: str = "gpt-3.5-turbo"
+    main_score: str = f"{model_name}_judge"
+
+    reduction_map: Dict[str, List[str]] = None
+    openai_models: List[str] = ["gpt-3.5-turbo"]
+    anthropic_models: List[
+        str
+    ] = []  # this is here for the sake of documentation for future models
+    external_api_models = openai_models + anthropic_models
+
+    _requirements_list: List[str] = ["llama_index"]
+
+    def _model_using_extrnal_api(self):
+        return self.model_name not in self.external_api_models
+
+    def prepare(self):
+        """Initialization method for the metric. Initializes the CorrectnessEvaluator with the OpenAI model."""
+        super().prepare()
+
+        self.reduction_map: Dict[str, List[str]] = {"mean": [self.main_score]}
+
+        from llama_index.core.evaluation import CorrectnessEvaluator
+
+        assert (
+            (not self._model_using_extrnal_api)
+            or (settings.allow_passing_data_to_remote_api)
+        ), f"Cannot run expression by {self} when unitxt.settings.allow_passing_data_to_remote_api=False either set it to True or set UNITXT_ALLOW_PASSING_DATA_TO_REMOTE_API environment variable."
+
+        if self.model_name in self.openai_models:
+            from llama_index.llms.openai import OpenAI
+
+            llm = OpenAI("gpt-3.5-turbo")
+        else:
+            raise NotImplementedError(
+                f"LlamaIndexCorrectnessMetric does not support {self.model_name}, currently only gpt-3.5-turbo is supported"
+            )
+
+        self.evaluator = CorrectnessEvaluator(llm=llm)
+
+    def compute(
+        self,
+        references: List[List[Any]],
+        predictions: List[Any],
+        additional_inputs: List[Dict],
+    ) -> List[Dict[str, Any]]:
+        """Method to compute the correctness metric.
+
+        Args:
+            references (List[List[Any]]): List of reference instances.
+            predictions (List[Any]): List of predicted instances.
+            additional_inputs (List[Dict]): List of additional input data.
+
+        Returns:
+            List[Dict[str, Any]]: List of computed scores and feedback.
+
+        Raises:
+            AssertionError: If the input does not meet the expected format.
+        """
+        # treat the references as the questions and the predictions as answers
+        # assume a single reference
+
+        response_list = predictions
+
+        query_list = [instance["question"] for instance in additional_inputs]
+        contexts_list = [instance["contexts"] for instance in additional_inputs]
+
+        references_responses_list = references
+
+        results = []
+        for query, response, contexts, reference_responses in zip(
+            query_list, response_list, contexts_list, references_responses_list
+        ):
+            per_instance_results = []
+            for reference_response in reference_responses:
+                per_instance_results.append(
+                    self.evaluator.evaluate(
```

Comment:
If you are running the evaluator one instance at a time - why use BulkInstanceMetric and not use an InstanceMetric?  It would remove need for the loops.

Reply:
nice, didn't know about this, will fix

Reply:
done